libraries/Microsoft.Bot.Expressions/ExpressionFunctions.cs

```diff
@@ -3748,6 +3748,14 @@ private static int GetPropertyCount(dynamic obj)
                     ValidateIsMatch),
             };
 
+            var eval = new ExpressionEvaluator(ExpressionType.Optional, null, ReturnType.Boolean, ExpressionFunctions.ValidateUnaryBoolean);
```

Comment:
>null [](start = 72, length = 4)

Should be a delegate that returns the underlying expression value.

Reply:
This is exactly the code you had before:
   ExpressionEvaluator eval;
            if (type == Optional || type == Ignore)
            {
                eval = new ExpressionEvaluator(type, null, ReturnType.Boolean, BuiltInFunctions.ValidateUnaryBoolean);
                eval.Negation = eval;
            }

For both Optional and eval you had null passed in.

Am I missing something?

---
In reply to: [376585872](https://github.com/microsoft/botbuilder-dotnet/pull/3350#discussion_r376585872) [](ancestors = 376585872)

Reply:
Does this look right?
            var eval = new ExpressionEvaluator(ExpressionType.Optional, (expression, state) => throw new NotImplementedException(), ReturnType.Boolean, ExpressionFunctions.ValidateUnaryBoolean);
            eval.Negation = eval;
            functions.Add(eval);
            
            eval = new ExpressionEvaluator(ExpressionType.Ignore, (expression, state) => expression.TryEvaluate(state), ReturnType.Boolean, ExpressionFunctions.ValidateUnaryBoolean);
            eval.Negation = eval;
            functions.Add(eval);


---
In reply to: [376615100](https://github.com/microsoft/botbuilder-dotnet/pull/3350#discussion_r376615100) [](ancestors = 376615100,376585872)